plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/JavaToJKTreeBuilder.kt

```diff
@@ -960,13 +967,27 @@ class JavaToJKTreeBuilder(
         }
 
         fun PsiLocalVariable.toJK(): JKLocalVariable {
+            val codeBlock = parentOfType<PsiCodeBlock>()
+            val mutability: Mutability
+            if (hasModifierProperty(PsiModifier.FINAL)) {
+                mutability = IMMUTABLE
+            } else {
+                val projectInstance = InspectionManager.getInstance(project)
+                val canBeFinal = LocalCanBeFinal().checkCodeBlockForImmutability(codeBlock, projectInstance)
+                mutability = when {
+                    canBeFinal?.any { it.toString() == "Variable '${name}' can have 'final' modifier" } == true -> IMMUTABLE
+
+                    canBeFinal != null || codeBlock == null -> UNKNOWN
+                    codeBlock.hasMatchingNameAssignment(this) -> MUTABLE
```

Comment:
This is not future-proof as there maybe declarations with the same name in the block. What's the reason behind this?